### ISSUE TITLE
Instructor 1.4.0 doesn't support the groq client due to the `.refusal` error

### ISSUE THREAD

**Issue by ivanleomk** [jxnl/instructor#953]
Groq is still using an older version of the OpenAI Sdk which doesn't have the `.refusal` parameter on the existing ChatCompetionMessage. 

As a result, the new integration with Structured Outputs breaks the groq integration when it comes to parsing the completed message.

This minimal reproduction replicates this error

```python
from instructor import from_groq
from pydantic import BaseModel
from groq import Groq


class User(BaseModel):
    name: str
    email: str
    age: int


client = from_groq(Groq())

resp = client.chat.completions.create(
    model="llama3-70b-8192",
    messages=[
        {
            "role": "user",
            "content": "Extract the user from this : Ivan is a 27 year old softrware engineer in Singapore",
        }
    ],
    response_model=User,
)

print(resp)
```

We can then derive the error message of

```bash
          ^^^^^^^^^^^^^^^^^
  File "/Users/ivanleo/Documents/coding/instructor/instructor/process_response.py", line 144, in process_response
    model = response_model.from_response(
            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/ivanleo/Documents/coding/instructor/instructor/function_calls.py", line 149, in from_response
    return cls.parse_tools(completion, validation_context, strict)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/ivanleo/Documents/coding/instructor/instructor/function_calls.py", line 318, in parse_tools
    message.refusal is None
    ^^^^^^^^^^^^^^^
  File "/Users/ivanleo/Documents/coding/instructor/venv/lib/python3.11/site-packages/pydantic/main.py", line 828, in __getattr__
    raise AttributeError(f'{type(self).__name__!r} object has no attribute {item!r}')
AttributeError: 'ChoiceMessage' object has no attribute 'refusal'
```

**Comment by S4mpl3r** [comment]
Also encountering this issue.
@ivanleomk While waiting for a fix, you can use `from_groq(Groq(), mode=instructor.Mode.JSON)` to bypass the error.

**Comment by ivanleomk** [comment]
with @callmephilip's new fix in #962 , this should be fixed in the next release @S4mpl3r ! :)
